### PR TITLE
[IND-510] setup containerized Docker env for e2e testing

### PR DESCRIPTION
**Pull request by dydxwill**
### Changelist
create a containerized Docker environment running the protocol locally with fixed prices for 3 markets (btc, eth, sol), Indexer, and all Indexer dependencies.

### Test Plan
ran locally

### Author/Reviewer Checklist
- [ ] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [ ] If the PR has breaking postgres changes to the indexer add the `indexer-postgres-breaking` label.
- [ ] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [ ] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [ ] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [ ] Manually add any of the following labels: `refactor`, `chore`, `bug`.

### CODE REVIEW

**Review by vincentwschau — APPROVED** [review]

File: e2e-testing/docker-compose-e2e-test.yml [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+    environment:
+      - DAEMON_HOME=/dydxprotocol/chain/.alice
+    volumes:
+      - ../protocol/localnet/dydxprotocol0:/dydxprotocol/chain/.alice/data
+    ports:
+      - "26657:26657"
+      - "9090:9090"
+      - "1317:1317"
+  
+  dydxprotocold1:
```
Comment (vincentwschau) [inline e2e-testing/docker-compose-e2e-test.yml:71]: nit: Comment that this is the indexer full-node.

File: e2e-testing/docker-compose-e2e-test.yml [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+        condition: service_completed_successfully
+  comlink:
+    build:
+      context: ../indexer
+      dockerfile: ../indexer/Dockerfile.service.local
+      args:
+        service: comlink
+        NPM_TOKEN: ${NPM_TOKEN}
+    environment:
+      - TENDERMINT_WS_URL=host.docker.internal:26657 # connects to localhost:26657 on host machine
```
Comment (vincentwschau) [inline e2e-testing/docker-compose-e2e-test.yml:170]: nit: No longer needed.

File: e2e-testing/docker-compose-e2e-test.yml [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+        condition: service_healthy
+      postgres-package:
+        condition: service_completed_successfully
+  comlink:
+    build:
+      context: ../indexer
+      dockerfile: ../indexer/Dockerfile.service.local
+      args:
+        service: comlink
+        NPM_TOKEN: ${NPM_TOKEN}
```
Comment (vincentwschau) [inline e2e-testing/docker-compose-e2e-test.yml:168]: nit: No longer needed.

File: e2e-testing/docker-compose-e2e-test.yml [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+       - cosmovisor
+       - run
+       - start
+       - --log_level
+       - error
+       - --home
+       - /dydxprotocol/chain/.bob
+       - --p2p.persistent_peers 
+       - "17e5e45691f0d01449c84fd4ae87279578cdd7ec@dydxprotocold0:26656,b69182310be02559483e42c77b7b104352713166@dydxprotocold1:26656,47539956aaa8e624e0f1d926040e54908ad0eb44@dydxprotocold2:26656,5882428984d83b03d0c907c1f0af343534987052@dydxprotocold3:26656"
+       - --non-validating-full-node=true
```
Comment (vincentwschau) [inline e2e-testing/docker-compose-e2e-test.yml:83]: nit: Does this work? In your `local.sh` `bob` is given some stake and is a validator, but the node here is set to be a full-node? Perhaps it's better to create a separate full-node for the indexer? Non-blocking if this works for now, but maybe a comment on how this works.

**Review by Christopher-Li — APPROVED** [review]

File: e2e-testing/docker-compose-e2e-test.yml [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+      DATADOG_POSTGRES_PASSWORD: dydxserver123
+    healthcheck:
+      test: ["CMD-SHELL", "pg_isready -U dydx_dev"]
+      interval: 5s
+      timeout: 20s
+      retries: 10
+  redis:
+    image: redis:5.0.6-alpine
+    ports:
+      - 6382:6379
```
[the remark this answers is not in the document]
Comment (dydxwill) [reply e2e-testing/docker-compose-e2e-test.yml:45]: yeah I can investigate this later.

**Pull request merged**